### PR TITLE
CMake changes (built in windows and linux) 

### PR DESCRIPTION
**Pull request by gel**
1) Fixes for OpenCV > 2.4.3
2) Update to newest eigen3 cmake file (supports environment variable).

### CODE REVIEW

File: examples/CMakeLists.txt [hunk 1]
```diff
@@ -7,10 +7,10 @@ cmake_minimum_required(VERSION 2.8.10)
 #version variable is defined from project root's CMakeLists
 if("${OpenCV_VERSION_MAJOR}$" EQUAL 2)
   message(STATUS "OpenCV 2.x detected")
-  find_package(OpenCV 2.4.3 REQUIRED core imgproc highgui)
```
Comment (patrikhuber) [inline examples/CMakeLists.txt]: Do you think we can leave 2.4.3 here, or does that cause problems as well? My thought is that if we just put "2", then CMake will not fail if someone tries to run it with OpenCV 2.3, 2.2 or even older, but eos will most likely not work with such an old version.

**Comment by patrikhuber** [comment]
Thank you very much for the PR! Very good changes. In fact both of these I already included in the devel branch (which contains a major CMake modernisation), but I'm happy if we merge these small fixes into the master branch now.

**Comment by gel** [comment]
I wasn't aware of this branch - I will take a look.

The correct fix is indeed OpenCV 2 / OpenCV 3.
It needs the same change in the auxiliary projects / cmakelists as well.

File: examples/CMakeLists.txt (hunk 1 above)
Comment (gel) [inline examples/CMakeLists.txt]: works.

**Comment by patrikhuber** [comment]
> The correct fix is indeed OpenCV 2 / OpenCV 3.

I'm slightly confused now as you commented that changing "2" back to "2.4.3" works as well (and it works with my 2.4.12 installation). What do you mean by above?

> I wasn't aware of this branch - I will take a look.

It's mainly CMake modernisation, and some improvements to the Python/Matlab bindings. No changes to the actual fitting algorithm(s). I guess it would be time to merge devel into master, I wanted to get some experience with it first and verify whether everything works, and it all works great so far. I'll try to merge over the next days.

**Comment by gel** [comment]
Sorry for the confusion, let me try to clarify the problems I encountered.

I have only OpenCV 3.1 installed in my windows environment.
find_package(OpenCV REQUIRED core) - CMake won't work when given specific
version number, or partial version number (3 won't match with 2).

In all the places where you have the checks of OpenCV 2 and OpenCV 3 - it
works with both partial or full version number.

**Comment by patrikhuber** [comment]
Ok, cool, thanks for the clarification. This coincides with what I've observed.

Thanks again for contributing!

**Pull request merged**